test/real_content/test_swupd.bats

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env bats
+
+load "../functional/testlib"
+
+# Test swupd with real official content published on clearlinux.org
+# This test assumes all functional tests to be passing in order to work,
+# because we aren't going to test all swupd functionalities. The idea is
+# to test major swupd commands and standard use cases
+
+#TODO: Add test for crossing a format barrier
+
+# Optional parameters
+# $URL: Clearlinux content and version urls
+# $MAX_VERSIONS: The maximum number of versions we are going to be looking to test updates. Should be # at least 2 (current + update). Default is 15.
+
+local SWUPD_OPTS # Swupd options to use in tests
+#local FORMAT # Format of the last available version
+local VERSIONS # Array of last available versions. First item is the oldest
+local ROOT_DIR # Root directory of the installation
+
+# get_format: Echos the format of the version in $1
+# TODO: This is not the best way to handle that. we should have in swupd
+# codebase the information about the supported format of this swupd.
+get_format() {
+	v="$1"
+	curl -f ${URL}/${v}/format 2>/dev/null
+	return 0
+}
+
+# get_format_from_versions: Echos the first format available in a version from
+# a list of versions
+get_format_from_versions() {
+	for v in "$@"; do
+		format=$(get_format $v)
+		if [ -z "$format" ]; then # if version isn't published, discard
+			continue
+		else
+			echo $format
+			break
+		fi
+	done
+}
+
+test_setup() {
+	# Fill defaults in optional parameters
+	if [ -z "$URL" ]; then
+		URL=https://cdn.download.clearlinux.org/update
+	fi
+	if [ -z "$MAX_VERSIONS" ]; then
+		MAX_VERSIONS=15
+	fi
+
+	# Using github clr-bundles project to get official clear releases
+	#TODO: Discover available versions to work with custom mixers
+	#Depends on: https://github.com/clearlinux/mixer-tools/issues/467
+	version_list=$(git ls-remote --tags https://github.com/clearlinux/clr-bundles.git | grep -v {} | grep -v latest | sed "s/.*refs\/tags\///" | sort -gr | head -n $MAX_VERSIONS)
+	if [ -z "$version_list" ]; then
+		echo "Impossible to get versions list"
+		return 1
+
+	fi
+
+	# Get the last format
+	export FORMAT=$(get_format_from_versions $version_list)
+	version_list=$(echo "$version_list" | sort -g)
+	if [ -z "$FORMAT" ]; then
+		echo "Impossible to guess format"
+		return 1
+
+	fi
+
+	# Fill a list of last available versions (up to ${MAX_VERSIONS})
+	i=0
+	for v in $version_list; do
+		format=$(get_format $v)
+		# if version isn't published or is from a different version, discard
+		if [ -z "$format" ] || [ "$FORMAT" -ne "${format}" ]; then
+			continue
+		fi
+
+		VERSION[$i]=$v
+		i=$(($i+1))
```

Comment:
Other test files omit the leading `$` for variable references in arithmetic expressions. For consistency, you can write this as `i=$((i+1))`.

The same comment applies to several other places in this source file, so I won't comment on those lines.